src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpCodeActionAndSolutionGenerator.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PullMemberUp;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp
+{
+    [ExportLanguageService(typeof(IPullMemberUpActionAndSolutionGenerator), LanguageNames.CSharp), Shared]
+    internal class CSharpCodeActionAndSolutionGenerator : AbstractCodeAndSolutionGenerator
+    {
+        protected override void RemoveNode(DocumentEditor editor, SyntaxNode node, ISymbol symbol)
+        {
+            if (node is VariableDeclaratorSyntax variableDeclarator &&
+                (symbol.Kind == SymbolKind.Field || symbol.Kind == SymbolKind.Event))
+            {
+                if (variableDeclarator.Parent != null &&
+                    variableDeclarator.Parent.Parent is BaseFieldDeclarationSyntax fieldOrEventDeclaration)
+                {
+                    if (fieldOrEventDeclaration.Declaration.Variables.Count() == 1)
+                    {
+                        // If there is only one variable, e.g.
+                        // public int i = 0;
+                        // Just remove all
+                        editor.RemoveNode(fieldOrEventDeclaration);
+                    }
+                    else
+                    {
+                        // If there are multiple variables, e.g.
+                        // public int i, j = 0;
+                        // Remove only one variable
+                        editor.RemoveNode(variableDeclarator);
+                    }
+                }
```

Comment:
you can simplify this by just doing: `editor.RemoveNode(editor.Generator.GetDeclaration(node))`.  This should be doable from the base class.